### PR TITLE
Drop Metadata protocol support. No longer needed

### PR DESCRIPTION
**Pull request by adepue**
## Description

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/azure/walinuxagent/1752)

### CODE REVIEW

File: azurelinuxagent/common/protocol/util.py [hunk 1]
```diff
@@ -319,8 +309,7 @@ def clear_protocol(self):
 
     def get_protocol(self, by_file=False):
         """
-        Detect protocol by endpoints, if by_file is True,
```
Comment (narrieta) [inline azurelinuxagent/common/protocol/util.py]: let's remove the parameter as well (by_file)

Comment (narrieta) [inline azurelinuxagent/common/protocol/util.py]: we need to remove the unit tests as well (test_metadata.py)

File: azurelinuxagent/common/protocol/util.py [hunk 2]
```diff
@@ -57,7 +56,7 @@ def __getattr__(self, name):
         raise AttributeError("%s not a valid value" % name)
 
 
-prots = _nameset(("WireProtocol", "MetadataProtocol"))
+prots = _nameset(("WireProtocol", ))
```
Comment (vrdmr) [inline azurelinuxagent/common/protocol/util.py]: nit; Please remove the trailing ",".

File: azurelinuxagent/common/protocol/util.py (hunk 1 above)
Comment (vrdmr) [reply azurelinuxagent/common/protocol/util.py]: https://travis-ci.org/Azure/WALinuxAgent/jobs/635512763 would give you the places metadata is used. Would help in removing the metadata protocol based unittests.

File: azurelinuxagent/common/protocol/util.py (hunk 2 above)
Comment (adepue) [reply azurelinuxagent/common/protocol/util.py]: The trailing , ensures that a tuple is created... I tried to preserve the data shape to ensure that we didn't get weird values downstream from the type change.  Changing the type will cause it to enumerate as a string correct and break nameset up into letters...  Or maybe I mis tested that.

File: azurelinuxagent/common/protocol/util.py (hunk 1 above)
Comment (adepue) [reply azurelinuxagent/common/protocol/util.py]: Done, as well as mockmetadata.py

Comment (adepue) [reply azurelinuxagent/common/protocol/util.py]: Done

Comment (vrdmr) [reply azurelinuxagent/common/protocol/util.py]: Searching for azurelinuxagent.common.protocol.metadata would point you to other places where metadata is used.

**Comment by narrieta** [comment]
Running automation

**Comment by adepue** [comment]
Tests are cleaned up now.  There are some lingering pieces around "detecting" the metadata protocol/etc via tag file, but I need to spend more time looking at those to make sure I don't remove too much.  Right now this code ensures either the wire server is detected and returned, or exceptions are raised.

File: azurelinuxagent/common/protocol/util.py (hunk 1 above)
Comment (narrieta) [reply azurelinuxagent/common/protocol/util.py]: there is a call in that uses the parameter in azurelinuxagent/pa/provision/default.py

Comment (adepue) [reply azurelinuxagent/common/protocol/util.py]: Cleaned up in latest commit

**Comment by narrieta** [comment]
Automation failing: 

> unable to read wireserver ip: [Errno 2] No such file or directory: '/var/lib/waagent/WireServerEndpoint'

**Comment by narrieta** [comment]
Replaced by #1780

**Pull request closed without merging**